### PR TITLE
feat(ingest): Allow excluding data when computing `search_message`

### PR DESCRIPTION
**Pull request by lobsterkatie**
During ingest, we add a `search_message` attribute to the event, which contains various values that we can match on when searching or when applying an alert rule - things like the title, culprit, etc. It also contains values from the event's `metadata` attribute. Not all values we might want to put there are actually helpful for searching and rule matching, though, so this adds the ability to exclude certain metadata values from being added to the string. It also skips adding booleans and numbers, since those aren't meaningful without their keys.

### CODE REVIEW

File: src/sentry/eventstore/models.py [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
 # Keys in the event payload we do not want to send to the event stream / snuba.
 EVENTSTREAM_PRUNED_KEYS = ("debug_meta", "_meta")
+# Keys in the event metadata we do not want to include in the event's `search_message`
+SEARCH_MESSAGE_SKIPPED_KEYS = ()
```
Comment (wedamija) [inline src/sentry/eventstore/models.py:36]: Let's make this a frozenset?

File: src/sentry/eventstore/models.py [hunk 2]
```diff
@@ -554,7 +556,10 @@ def search_message(self) -> str:
             message += data["logentry"].get("formatted") or data["logentry"].get("message") or ""
 
         if event_metadata:
-            for value in event_metadata.values():
+            for key, value in event_metadata.items():
+                if key in SEARCH_MESSAGE_SKIPPED_KEYS or type(value) in [bool, int, float]:
```
Comment (wedamija) [inline src/sentry/eventstore/models.py:560]: `isinstance(value, (bool, int, float))`

**Review by wedamija — COMMENTED** [review]
What keys do we plan to use this with?

File: src/sentry/eventstore/models.py (hunk 1 above)
Comment (lobsterkatie) [reply src/sentry/eventstore/models.py:36]: Done

File: src/sentry/eventstore/models.py (hunk 2 above)
Comment (lobsterkatie) [reply src/sentry/eventstore/models.py:560]: Done

**Comment by lobsterkatie** [comment]
> What keys do we plan to use this with?

I'm adding a metric for Armen about whether, when we create a new group, the stacktrace is all in-app, all system, or a mix. That's determined when we run the grouping algorithms, and then the information travels (in `metadata`) to the spot at which we decide whether or not to make a new group, but aving one of `all-in-app`, `all-system`, and `mixed` included in the search message is kind of a waste of space. (If we really wanted to be able to search on that, we should include it as a tag, which we're not currently choosing to do.)

**Review by wedamija — APPROVED** [review]

**Pull request merged**